### PR TITLE
Download_Model_at_ImageBol_Run_time

### PR DESCRIPTION
**Pull request by ravi5175**
added function to download model at runtime from the official freshly-built repository
added a better progress bar for visualizing downloading progress
added dictionary structure for handling typo file errors
Personalized status codes for error handling and more error handling provided
the choice for large size model download for saving data
create a directory for models
check the existence of model in the directory
cleaning of the corrupted file in case of download interrupt

tested on:
official freshly-built repository
stable internet connection

warning:
unstable internet connection might cause the function to halt unexpectedly

### CODE REVIEW

**Comment by KapilBansal320** [comment]
Duplicate of #5

File: freshlybuiltimagebol/downloader.py [hunk 1]
```diff
@@ -6,40 +6,45 @@
 
 model_url= "https://raw.githubusercontent.com/FreshlyBuilt/freshlybuiltimagebol/master/freshlybuiltimagebol/models/"
 
+available_models={"F_est":"frozen_east_text_detection"}
+
```
Comment (vishal2612200) [inline freshlybuiltimagebol/downloader.py:10]: Add Class to the Structure.

File: freshlybuiltimagebol/downloader.py [hunk 2]
```diff
@@ -1,52 +1,52 @@
...
+        
+                            if total_length is None:
+                                f.write(response.content)
+                            else:
+                                chunk_size=1024
+                                for data in tqdm(iterable = response.iter_content(chunk_size),total=total_length/chunk_size, unit = 'KB'):
+                                    f.write(data)
+                                tqdm.close()
+                        if total_length != 0 and tqdm.n != total_length:
+                            print("ERROR, something went wrong")
```
Comment (vishal2612200) [inline freshlybuiltimagebol/downloader.py:37]: Add animation and dist functionality to original code

File: freshlybuiltimagebol/downloader.py [hunk 3]
```diff
@@ -1,52 +1,52 @@
...
+
+
+
+
+class img_bol_downloader:
+
+    def download_model(model_name):
+        available_models={"F_est":"frozen_east_text_detection"}
+        model_url= "https://raw.githubusercontent.com/FreshlyBuilt/freshlybuiltimagebol/master/freshlybuiltimagebol/models/"
+        if model_name in available_models:
```
Comment (vishal2612200) [inline freshlybuiltimagebol/downloader.py:14]: output_dir="freshlybuiltimagebol/models/"+model_name+".pb"
		if isfile(output_dir):
			print("Model already exists")

Add this use case to your function

**Pull request merged**